instrumentation/kafka-streams/src/main/java/brave/kafka/streams/TracingProcessorSupplier.java

```diff
@@ -0,0 +1,63 @@
+package brave.kafka.streams;
+
+import brave.Span;
+import brave.Tracer;
+import org.apache.kafka.streams.processor.Processor;
+import org.apache.kafka.streams.processor.ProcessorContext;
+import org.apache.kafka.streams.processor.ProcessorSupplier;
+
+class TracingProcessorSupplier<K, V> implements ProcessorSupplier<K, V> {
+
+  final KafkaStreamsTracing kafkaStreamsTracing;
+  final Tracer tracer;
+  final String name;
+  final Processor<K, V> delegateProcessor;
+
+  TracingProcessorSupplier(KafkaStreamsTracing kafkaStreamsTracing,
+      String name,
+      Processor<K, V> delegateProcessor) {
+    this.kafkaStreamsTracing = kafkaStreamsTracing;
+    this.tracer = kafkaStreamsTracing.tracing.tracer();
+    this.name = name;
+    this.delegateProcessor = delegateProcessor;
+  }
+
+  /** This wraps transform method to add tracing. */
+  @Override public Processor<K, V> get() {
+    return new Processor<K, V>() {
+      ProcessorContext processorContext;
+
+      @Override
+      public void init(ProcessorContext processorContext) {
+        this.processorContext = processorContext;
```

Comment:
add a comment that `init()` is done on the same thread as `supplier.get()` and which code in kafka does that..  assuming that's the case.

Reply:
I've found that this does not happen:

ProcessorSupplier docs:
```
 * It is used in {@link Topology} for adding new processor operators, whose generated
 * topology can then be replicated (and thus creating one or more {@link Processor} instances)
 * and distributed to multiple stream threads.
```

Then, a new processor is created on potentially different threads as the `get` method. Should I then pass the tracer on the processor constructor, right? I'm planning to move the anonymous processor/transformer to a class `TracingProcessor/Transformer`:

On processorSupplier:
```
  @Override public Processor<K, V> get() {
    return new TracingProcessor<>(kafkaStreamsTracing, name, delegateProcessor);
  }
```
and `TracingProcessor`:
```
  public TracingProcessor(KafkaStreamsTracing kafkaStreamsTracing,
      String name, Processor<K, V> delegateProcessor) {
    this.kafkaStreamsTracing = kafkaStreamsTracing;
    this.tracer = kafkaStreamsTracing.tracing.tracer();
    this.name = name;
    this.delegateProcessor = delegateProcessor;
  }
```
Is this mechanism enough to trace correctly each processor instance?